File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/Export/exporttoxml.rb
module Asposediagramjava
  module ExportToXml
    def initialize()
        # Exporting VSD to VDX
        export_to_vdx()

        # Exporting VSD to VSX
        export_to_vsx()

        # Exporting VSD to VTX
        export_to_vtx()
    end

    def export_to_vdx()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Call the diagram constructor to load diagram from a VSD file
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        # Save as VDX
        diagram.save(data_dir + "Diagram.vdx", Rjb::import('com.aspose.diagram.SaveFileFormat').VDX)

        puts "Exported visio diagram to VDX."
    end

    def export_to_vsx()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Call the diagram constructor to load diagram from a VSD file
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        # Save as VSX
        diagram.save(data_dir + "Diagram.vsx", Rjb::import('com.aspose.diagram.SaveFileFormat').VSX)

        puts "Exported visio diagram to VSX."
    end

    def export_to_vtx()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Call the diagram constructor to load diagram from a VSD file
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        # Save as VTX
        diagram.save(data_dir + "Diagram.vtx", Rjb::import('com.aspose.diagram.SaveFileFormat').VTX)

        puts "Exported visio diagram to VTX."
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/LoadingSavingandConverting/ExportToXml.php
<?php
namespace Aspose\Diagram\LoadingSavingandConverting;

use com\aspose\diagram\Diagram as Diagram;
use com\aspose\diagram\SaveFileFormat as SaveFileFormat;
class ExportToXml{

    public static function run($dataDir=null){

        # Exporting VSD to VDX
        ExportToXml::export_to_vdx($dataDir);

        # Exporting VSD to VSX
        ExportToXml::export_to_vsx($dataDir);

        # Exporting VSD to VTX
        ExportToXml::export_to_vtx($dataDir);
    }

    public static function export_to_vdx($dataDir=null){

        # Call the diagram constructor to load diagram from a VSD file
        $diagram = new Diagram($dataDir."drawing.vsd");

        # Save as VDX
        $saveFileFormat=new SaveFileFormat();
        $diagram->save($dataDir."Diagram.vdx", $saveFileFormat->VDX);

        print "Exported visio diagram to VDX.".PHP_EOL;
    }

    public static function export_to_vsx($dataDir=null){

        # Call the diagram constructor to load diagram from a VSD file
        $diagram = new Diagram($dataDir."drawing.vsd");

        # Save as VSX
        $saveFileFormat=new SaveFileFormat();
        $diagram->save($dataDir."Diagram.vsx", $saveFileFormat->VSX);

        print "Exported visio diagram to VSX.".PHP_EOL;
    }

    public static function export_to_vtx($dataDir=null){

        # Call the diagram constructor to load diagram from a VSD file
        $diagram = new Diagram($dataDir."drawing.vsd");

        # Save as VTX
        $saveFileFormat=new SaveFileFormat();
        $diagram->save($dataDir."Diagram.vtx", $saveFileFormat->VTX);

        print "Exported visio diagram to VTX.".PHP_EOL;
    }
}